core/src/main/java/org/apache/druid/guice/GuiceAnnotationIntrospector.java

```diff
@@ -101,13 +103,28 @@ public Object findInjectableValueId(AnnotatedMember m)
     // delegate creators. I'm not 100% sure why it's not called, but guess it's because the argument
     // is some Java type that Jackson already knows how to deserialize. Since there is only one argument,
     // Jackson perhaps is able to just deserialize it without introspection.
+
     if (ac instanceof AnnotatedParameter) {
       final AnnotatedParameter ap = (AnnotatedParameter) ac;
       if (ap.hasAnnotation(JsonProperty.class)) {
         return JsonIgnoreProperties.Value.empty();
       }
     }
 
-    return JsonIgnoreProperties.Value.forIgnoredProperties("");
+    // A map can have empty string keys e.g. https://github.com/apache/druid/issues/10859. By returning empty ignored
+    // list for map, we can allow for empty string keys in a map. A nested class within map
+    // can still be annotated with JacksonInject and still be non-deserializable from user input
+    // Refer to {@link com.fasterxml.jackson.databind.deser.BasicDeserializerFactory.createMapDeserializer} for details
+    // on how the ignored list is passed to map deserializer
+    if (ac instanceof AnnotatedClass) {
+      final AnnotatedClass aClass = (AnnotatedClass) ac;
+      if (Map.class.isAssignableFrom(aClass.getAnnotated())) {
+        return JsonIgnoreProperties.Value.empty();
+      }
+    }
+
+    // We will allow serialization on empty properties. Properties marked with {@code @JacksonInject} are still
+    // not serialized if there is no getter marked with {@code @JsonProperty}
```

Comment:
Do we have any class that has a field injected with `@JacksonInject` and a getter method with `@JsonProperty` for that field? It seems pretty strange if we have any.

Reply:
There is none. which is why it's ok to allow getters otherwise we might have serialized fields marked with @JacksonInject.